### PR TITLE
Add custom fields and log

### PR DESCRIPTION
**Pull request by debiff**
## THIS PR DEPENDS ON #9 
This PR add the functions that allows to add a log and the custom fields.

### CODE REVIEW

File: ios/RNZendeskChat.m [hunk 1]
```diff
@@ -69,14 +60,46 @@ @implementation RNZendeskChat
...
+        mutableLog = [[NSMutableString alloc] init];
+    }
+    if(logId == nil){
+        logId = [NSNumber numberWithLong: 4413278012049];
+    }
+}
+RCT_EXPORT_METHOD(appendLog:(NSString *)log) {
+    [self initGlobals];
+    [mutableLog insertString:log atIndex:0];
+    [mutableLog substringToIndex:fmax(0,fmin(MAX_LOG_LENGTH,mutableLog.length))];
```
Comment (Undermaken) [inline ios/RNZendeskChat.m]:
```suggestion
    [mutableLog substringToIndex:fmax(0,fmin(MAX_LOG_LENGTH,mutableLog.length - 1))];
```

File: ios/RNZendeskChat.m [hunk 2]
```diff
@@ -69,14 +60,46 @@ @implementation RNZendeskChat
...
+}
+- (void) addTicketCustomFieldFunction:(NSString *)key withValue:(NSString *)value
+{
+    if(customFields == nil){
+        customFields = [[NSMutableDictionary alloc] init];
+    }
```
Comment (Undermaken) [inline ios/RNZendeskChat.m]:
```suggestion
    if(customFields == nil){
        customFields = [[NSMutableDictionary alloc] init];
    }
```
Maybe could we move this init `initGlobals`?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ios/RNZendeskChat.m (hunk 1 above)
Comment (debiff) [reply ios/RNZendeskChat.m]: Thank you, fixed in [eafe3a3](https://github.com/pagopa/io-react-native-zendesk/pull/10/commits/eafe3a3188141338377478de3f55ab2380d8ccb1)

File: ios/RNZendeskChat.m (hunk 2 above)
Comment (debiff) [reply ios/RNZendeskChat.m]: Yes, it's a good idea! Moved in [ab03762](https://github.com/pagopa/io-react-native-zendesk/pull/10/commits/ab03762f7e313302a55bb06cd58402ff1ab1d319)

**Review by Undermaken — APPROVED** [review]

**Pull request merged**